receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -0,0 +1,70 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package githubscraper // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/gitproviderreceiver/internal/scraper/githubscraper"
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/gitproviderreceiver/internal/metadata"
+)
+
+var (
+	errClientNotInitErr = errors.New("http client not initialized")
+)
+
+type githubScraper struct {
+	client   *http.Client
+	cfg      *Config
+	settings component.TelemetrySettings
+	logger   *zap.Logger
+	mb       *metadata.MetricsBuilder
+}
+
+func (ghs *githubScraper) start(_ context.Context, host component.Host) (err error) {
+	ghs.logger.Sugar().Info("Starting the scraper inside scraper.go")
```

Comment:
Do you have the scraper name already as part of the logger context? If not, include the scraper name. And the file name seems outdated. From a UX perspective, I would write something like:

```suggestion
	ghs.logger.Info("Starting the GitHub scraper", zap.String("host", ghs.cfg.Endpoint))
```

Reply:
This was something I added when I was first learning how to develop a receiver. I agree, it's message can either be removed or improved in general to be more informative to the user reading the logs.

Reply:
I've updated the message to just say `starting the GitHub scraper`. Right now the library making the calls to GitHub presumes publicly available github instead of hosted, so Endpoint isn't something provided by default. Will adjust this message accordingly in future iterations for clarity as I additionally create test cases for privately hosted enterprise github via custom endpoints.